client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -51,6 +51,7 @@ import { retargetViewPlans } from 'calypso/lib/analytics/ad-tracking';
 import { planItem as getCartItemForPlan } from 'calypso/lib/cart-values/cart-items';
 import { getPlanFeaturesObject } from 'calypso/lib/plans/features-list';
 import scrollIntoViewport from 'calypso/lib/scroll-into-viewport';
+import { usePlanUpgradeCreditsDisplay } from 'calypso/my-sites/plan-features-2023-grid/hooks/use-plan-upgrade-credits-display';
```

Comment:
Just noticed that `usePlanUpgradeCreditsDisplay` is missing any documentation. we probably want to add something, a brief description maybe. 

The discrepancy between that and `usePlanUpgradeCredits` I think should be clearer. In fact, considering that `usePlanUpgradeCredits` is only used in `usePlanUpgradeCreditsDisplay`, then I'm not sure if it should even be living and exported separately. It's a little confusing with both living side by side. Maybe just a single `usePlanUpgradeCredits` might do. anw - probably NIT.

Reply:
Yes you are correct, this is indeed confusing 😅 .
I think this mainly stems from the fact that my attempt here was to unify hooks and subsequently resulting bad naming on my part.
usePlanUpgradeCredits is used in several places and the main purpose of that hook is to calculate the plan upgrade credit. But `usePlanUpgradeCreditsDisplay` is complex UI logic that decides if the credits should be displayed. Maybe dividing these two hooks instead of unifying them and giving a proper name and some explanation should fix the confusion.

Reply:
I have changed the names to be more meaningful and added a comment.
https://github.com/Automattic/wp-calypso/blob/50a0b38f7bc047b728db47d9b580d2130f4c4bb1/client/my-sites/plans-features-main/components/plan-notice.tsx#L10-L11